dcos/http.py

```diff
@@ -167,9 +173,12 @@ def request(method,
         return response
     elif response.status_code == 401:
         if auth_token is not None:
-            msg = ("Your core.dcos_acs_token is invalid. "
-                   "Please run: `dcos auth login`")
-            raise DCOSAuthenticationException(msg)
+            if prompt_login:
+                msg = ("Your core.dcos_acs_token is invalid. "
+                       "Please run: `dcos auth login`")
+                raise DCOSAuthenticationException(msg)
+            else:
+                header_challenge_auth(dcos_url.geturl())
```

Comment:
we need to call `request` with the same parameters after this call. The reason it works for some commands is that we do checks before the actual call, but this won't be true for all commands.